Text about error messages [I18N-ACTION-1015]

aphillips opened the pull request:
This includes reorganizing the section on localization somewhat. See commit notes for details.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/55.html" title="Last updated on Dec 3, 2021, 5:25 PM UTC (ee54c4e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/55/6bb258c...aphillips:ee54c4e.html" title="Last updated on Dec 3, 2021, 5:25 PM UTC (ee54c4e)">Diff</a>


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/55.html" title="Last updated on Jan 20, 2022, 6:04 PM UTC (a550c0e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/55/6bb258c...aphillips:a550c0e.html" title="Last updated on Jan 20, 2022, 6:04 PM UTC (a550c0e)">Diff</a>

aphillips committed:
Add text for stock health warning about selection/truncation.[I18N-ACTION-982]

- add new requirement `#char_trunc_grapheme_boundary`
- add paragraph explaining relationship to DOM
- add example div

Shout out to: https://w3c.github.io/webauthn/#sctn-strings-truncation
Also related to i18n-activity#589

aphillips committed:
Merge remote-tracking branch 'w3c/gh-pages' into gh-pages

aphillips committed:
Add text from geolocation#50 about error/exception messages [I18N-ACTION-1015]

Harvested the text in my long comment on geolocation and converted it to
specdev format. Reorganized the section on localization. Added new
requirements.

xfq commented on index.html:
```diff
@@ -2670,33 +2666,51 @@ <h3>Localization</h3>
 	<p class="advisement">Specifications for APIs or protocols SHOULD define how the user's locale is determined (this is sometimes called <a href="#lang_negotiation">language negotiation</a>).</p>
 	</div>
-
+	
```
```suggestion

```

xfq commented on index.html:
```diff
@@ -2670,33 +2666,51 @@ <h3>Localization</h3>
+<section id="errors" class="subtopic">
+<h3>Errors and Exceptions</h3>
+
+<p>Protocols, APIs, and document formats sometimes provide a field to pass a human-readable error or exception message from a service to the caller in the form of a string. In general, and as <a href="#l10n_api_message_metadata">indicated above</a>, any natural language text conveying human readable messages or human readable content needs to be associated with language and direction metadata. Wh ...
```
```suggestion
<p>Protocols, APIs, and document formats sometimes provide a field to pass a human-readable error or exception message from a service to the caller in the form of a string. In general, and as <a href="#l10n_api_message_metadata">indicated above</a>, any natural language text conveying human-readable messages or human-readable content needs to be associated with language and direction metadata. Where this metadata is missing, the processing or display of the text might be compromised.</p>
```

xfq commented on index.html:
```diff
@@ -2670,33 +2666,51 @@ <h3>Localization</h3>
+
+<p>Protocols, APIs, and document formats sometimes provide a field to pass a human-readable error or exception message from a service to the caller in the form of a string. In general, and as <a href="#l10n_api_message_metadata">indicated above</a>, any natural language text conveying human readable messages or human readable content needs to be associated with language and direction metadata. Wh ...
+
+<p>Often the intention of the specification author in providing an error or exception message is to convey debugging information to a software developer. Specification authors sometimes assume that error or exception messages are not seen by end users; that software developers will prefer these messages to be unlocalized or appear in a specific language (usually English); or that there are other  ...
```
annecdotes -> anecdotes

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips committed:
[I18N-ACTION-1019] Address 'natural language' definition.

- Added a glossary section and added LTLI and CHARMOD-NORM definitions
  needed to it
- Added links to natural language throughout.
- Changed definitions to use `textual content` to match CHARMOD-NORM or
  as appropriate.
- Changed `lang_basics_1` wording to just refer to text and not require
  that someone read the text.
- Added `.definition` style from LTLI into local.css.

aphillips committed:
Merge branch 'gh-pages' of https://github.com/aphillips/bp-i18n-specdev into gh-pages

aphillips committed:
Fixed typo (thanks @xfq!)

aphillips commented on index.html:
```diff
@@ -427,11 +427,11 @@ <h3>Basic requirements</h3>
 
 	<div class="req" id="dir_paragraphs">
-	<p class="advisement">It must be possible to indicate base direction for each individual paragraph-level item of natural language text that will be read by someone.</p>
+	<p class="advisement">It must be possible to indicate base direction for each individual paragraph-level item of <a>natural language</a> text that will be read by someone.</p>
```
@r12a I'd like to remove the awkward ".. that will be read by someone", perhaps by changing 'natural language' here to 'textual content'. Thoughts?

aphillips committed:
Further work to address [I18N-ACTION-1019] ("natural language")
- Changed 'textual content' to 'localizable content'
- Fixed associated uses.
- Added items to the revision log.

aphillips committed:
Codify EPUB33's file naming restrictions into specdev guidelines. [I18N-ACTION-1097]

- Add section #file_naming
- Add reference to EPUB33
- Add requirements and text

Note that the code point list is sorted and reorganized from that in
EPUB33, but is not different.

aphillips committed:
Merge branch 'w3c:gh-pages' into gh-pages

xfq commented:
I'm confused. Is this PR about error messages or file naming restrictions?

xfq commented:
Ah, I just saw the email in member-i18n-core. Never mind.

aphillips committed:
I18N-ACTION-1097: Further edits.

aphillips committed:
Changes requested in telecon of 2022-01-20
- removed glossary subsection
- moved note on terminology to introduction and added a link to
  I18N-GLOSSARY
- replaced 'localizable content' with 'localizable text'
- changed title of working with file and path name section and working
  with errors and error messages section
- Added character name of U+E0001
- Added explanation of file and path name limits

aphillips committed:
Fixed one typo and updated/fixed the changelog.

aphillips commented:
I'm going to merge these changes based on our conversation of 2022-01-20. Note that some text should be reviewed again by the WG and this change depends on changes by Richard to the glossary.
